src/com/google/enterprise/adaptor/database/MetadataColumns.java

```diff
@@ -62,13 +60,14 @@
       log.fine("element: `" + e + "'");
       e = e.trim(); 
       String def[] = e.split(":", 2);
-      if (2 != def.length) {
-        String errmsg = "expected two parts separated by colon: `" + e + "'";
-        throw new InvalidConfigurationException(errmsg);
+      if (def.length == 1) {
+        String columnName = def[0].trim();
+        if (!columnName.isEmpty()) {
```

Comment:
As discussed, if def.length is 1 and columnName is empty, we got an empty entry like "foo,,bar".  The splitter on ',' already drops empty entries from the end.  This drops empty entries in the middle.

As far as invalid def.length == 2 entries, the columnName and/or the metadataKey would be empty.  I will treat an empty metadataKey as an identity mapping ("foo:" is the same as "foo"). An empty columnName here will get caught by verifyColumns in init().